code/modules/uplink/uplink_items.dm

```diff
@@ -118,9 +139,17 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	return pick(4;0.75,2;0.5,1;0.25)
 
 /datum/uplink_item/proc/purchase(mob/user, datum/component/uplink/U)
-	var/atom/A = spawn_item(item, user, U)
-	if(purchase_log_vis && U.purchase_log)
-		U.purchase_log.LogPurchase(A, src, cost)
+	//Spawn base items
+	for(var/i in 1 to spawn_amount)
```

Comment:
```suggestion
	for(var/i in 1 to spawn_amount)
```
Don't you need to specify the len? Or does BYOND do that implicitly?

Reply:
spawn_amount is a number